asreviewcontrib/dory/feature_extractors/transformer_embeddings.py

```diff
@@ -94,20 +147,23 @@ def fit(self, X, y=None):
         # for sentence-transformers, so return self
         return self
 
-    def fit_transform(self, X, y=None):
+    def transform(self, X, y=None):
         if self.verbose:
             print("Embedding text...")
 
-        embeddings = self._model.encode(
-            X, show_progress_bar=self.verbose, normalize_embeddings=self.normalize
-        )
+        embeddings = self._model.encode(X, show_progress_bar=self.verbose)
+        embeddings = self._to_numpy(embeddings)
 
-        if self.quantize:
-            embeddings = quantize_embeddings(embeddings, precision=self.precision)
-            if hasattr(embeddings, "numpy"):
-                embeddings = embeddings.numpy()
         return embeddings
 
+    def _to_numpy(self, arr):
+        """Ensure input is a NumPy array."""
+        if hasattr(arr, "numpy"):
+            return arr.numpy()
+        if not isinstance(arr, np.ndarray):
+            return np.array(arr)
+        return arr
```

Comment:
@J535D165 This function I do have some doubt about, whether it is actually necessary haha

Reply:
Actually, I think we do need it, since there are 6 encode overloads that all return different datatypes. Sorry for the ping!

Reply:
```python
hasattr(arr, "numpy"):
    return arr.numpy()
```
This part is, of course, very ugly. What kind of library implements this, haha

```python
        if not isinstance(arr, np.ndarray):
            return np.array(arr)
        return arr
```

Don't be too nice. `return np.array(arr)` is enough.